lib/OpenQA/Worker/Common.pm

```diff
@@ -368,8 +368,9 @@ sub call_websocket {
             }
             else {
                 $ws = undef;
-                if ($tx->completed) {
-                    call_websocket($ua_url->parse($tx->res->headers->location));
+                if (my $location_header = ($tx->completed ? $tx->res->headers->location : undef)) {
+                    print "Following ws redirection to: $location_header\n";
+                    call_websocket($ua_url->parse($location_header));
```

Comment:
For the record, Mojolicious got orders of magnitude faster overall when we started using temporary variables instead of repeating method calls back in the days. 😉 It's just a good rule of thumb to follow.